### PR TITLE
Support WSL for Golang

### PR DESCRIPTION
**Pull request by attiasas**
## Summary
Introduces `WslUtils` for Windows/WSL UNC and drive-path normalization so callers can derive Linux paths for `wsl.exe --cd`. Extends `GoDriver` with an opt-in constructor flag that, on Windows only, runs `go` through `wsl.exe --cd <linux-path> --exec …` so builds against WSL filesystems and `/mnt/...` layouts behave consistently.

## Changes
- **`build-info-extractor`**: New [`WslUtils`](build-info-extractor/src/main/java/org/jfrog/build/extractor/WslUtils.java) (`normalizePathStringForWsl`, `isWslPath`, `toLinuxPath`, `windowsLocalPathToWslMount`, distro helpers, `toWslLinuxCdPath`).
- **`build-info-extractor-go`**: [`GoDriver`](build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java) — existing four-arg constructor delegates to new five-arg form with `runGoThroughWsl` default `false`; `resolveGoExecutableForWsl`, `runsThroughWsl()`, WSL-aware `runCmd` wiring.
- **Tests**: [`WslUtilsTest`](build-info-extractor/src/test/java/org/jfrog/build/extractor/WslUtilsTest.java) (including `normalizePathStringForWsl`, `toWslLinuxCdPath` null/non-Windows cases, extended `\\?\` drive paths, edge cases); [`GoDriverTest`](build-info-extractor-go/src/test/java/org/jfrog/build/extractor/go/GoDriverTest.java) (`resolveGoExecutableForWsl` variants, `runsThroughWsl` OS gate).

## Notes
- WSL execution is opt-in and only active when `runGoThroughWsl` is true **and** the JVM reports Windows; non-Windows behavior matches the previous `GoDriver` setup.

-----

- [ ] All [tests](https://github.com/jfrog/build-info/actions/workflows/integrationTests.yml) have passed. If this feature is not already covered by the tests, new tests have been added.

### CODE REVIEW

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java [hunk 1]
```diff
@@ -29,13 +31,52 @@ public class GoDriver implements Serializable {
...
+    }
+
+    /**
+     * @param runGoThroughWsl when {@code true} on Windows, runs {@code go} via {@code wsl.exe --cd <linux-path> --exec ...}
+     *                        so Linux toolchains see WSL-backed projects and Windows temp dirs under {@code /mnt/...}.
+     */
+    public GoDriver(String executablePath, Map<String, String> env, File workingDirectory, Log logger,
+                    boolean runGoThroughWsl) {
+        this.runGoThroughWsl = runGoThroughWsl && SystemUtils.IS_OS_WINDOWS;
+        this.goExecutableForWsl = resolveGoExecutableForWsl(executablePath);
```
Comment (fluxxBot) [inline build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java]: nit: resolveGoExecutableForWsl() called unconditionally even when WSL disabled. Minor waste on every non-WSL construction. Consider:
  ▎ this.goExecutableForWsl = runGoThroughWsl ? resolveGoExecutableForWsl(executablePath) : null;

**Review by fluxxBot — APPROVED** [review]
Please check if backward compatibility is maintained. Rest looks good

File: build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java (hunk 1 above)
Comment (attiasas) [reply build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java]: done
